compile-release-tools

```diff
@@ -21,16 +21,21 @@ set -o pipefail
 RELEASE_TOOLS=(
   blocking-testgrid-tests
   kubepkg
-  release-notes
+  krel
 )
 
 setup_env() {
   echo "Setting up environment..."
 
-  if [ -n "${GOBIN:-}" ]
-  then
-    export PATH="${PATH}:${GOBIN}"
+  if [[ -z "${RELEASE_TOOL_BIN:-}" ]]; then
```

Comment:
nit:

Do we want to make `RELEASE_TOOL_BIN` a thing? Why not just rely on `GOBIN`/`GOPATH` and default them properly?

<details>
<summary>e.g. with a naive implementation like</summary>

```sh
setup_env() {
  local gobin="$( go env GOBIN )"
  local gopath=$( go env GOPATH )"

  if [ -n "$gobin" ] ; then
    export PATH="${PATH}:${gobin}
    return
  fi
  if [ -n "$gopath" ] ; then
    export GOBIN="${gopath}/bin"
    export PATH="${PATH}:${GOBIN}"
    return
  fi
  echo >&2 "could not determine GOBIN"
  return 1
}
```
</details>


The reason why I bring this up: this is introducing a third var that changes the behaviour of the script in a potentially non-obvious way. Not a big deal for this small thing, but in general I am questioning if we'd want to expand the public API "just because" and despite the fact that we already have knobs available. Just to be clear, I did/do this myself all the time, but I am trying to get better and more deliberate when changing the external interface of anything, esp. shared stuff.

Also, if you still think we'd want/need this var, can we call it `RELEASE_TOOLS_PATH`, `RELEASE_TOOLS_BINPATH`, `RELEASE_BIN_DIR`, ... or that like?